scraping/scraping/spiders/amazon.py

```diff
@@ -0,0 +1,70 @@
+from logging import getLogger
+from typing import Iterator
+
+from scrapy_splash import SplashJsonResponse, SplashRequest
+
+from ..splash import scroll_end_of_page_script
+from ._base import BaseSpider
+
+logger = getLogger(__name__)
+
+
+class AmazonSpider(BaseSpider):
+    name = "amazon"
+    allowed_domains = ["amazon.de"]
+
+    def parse_SERP(self, response: SplashJsonResponse) -> Iterator[SplashRequest]:
+        """
+        The `Scrapy` framework executes this method.
+
+        Yields:
+            Iterator[SplashRequest]: Requests that will be performed to scrap each product page
+        """
+        # Save HTML to database
+        self._save_SERP(response)
+        urls = response.css("div.a-row.a-size-base.a-color-base a::attr(href)").getall()
+        prices = response.css(
+            "div.a-row.a-size-base.a-color-base span.a-price-whole::text"
+        ).getall()
+        products = dict(zip(urls, prices))
+
+        # Yield request for each product
+        logger.info(f"Number of products to be scraped {len(products)}")
+        for key, value in products.items():
```

Comment:
Consider:

```python
for url, price in zip(urls, prices):
```

... and removing  `products = dict(zip(urls, prices))`. For the log output you could use just `len(urls)` for example. Further, IMHO using clearly named variables improves readability here.

Reply:
Changed in [9176ca5](https://github.com/calgo-lab/green-db/pull/65/commits/9176ca586316bdf65505000d0a1a7f8153b96283)